backend/note_taking/views/block_views.py

```
@@ -0,0 +1,68 @@
+
+class EditBlockEndpoint(UpdateAPIView):
+    serializer_class = BlockSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return Block.objects.filter(note__notebook__owner=self.request.user)
+
+    def perform_update(self, serializer):
+        block = serializer.instance
+        if block.note.notebook.owner.id != self.request.user.id:
+            raise PermissionDenied("You do not have permission to edit blocks in this note")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)